clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/ClusterCachingAgent.groovy

```diff
@@ -351,6 +355,18 @@ class ClusterCachingAgent implements CachingAgent, OnDemandAgent, AccountAware,
       .groupBy { it.autoScalingGroupName }
   }
 
+  private Map<String, List<Map>> loadLifecycleHooks(AmazonClients clients, List<String> autoScalingGroupNames) {
+    autoScalingGroupNames.collect { loadLifecycleHooks(clients, it) }*.groupBy { it.autoScalingGroupName } as Map<String, List<Map>>
```

Comment:
Yeah, there's no method of looking them up in bulk. This was extra code that's not actually necessary for this PR, I was adding it so we could expose it via other means later on. Agree, it'd be a lot of extra load on the API. I'll remove.